### PR TITLE
Properly resolve Mocha when using browser-runner

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Applies the fix described in https://github.com/webdriverio/webdriverio/issues/11003 to properly resolve Mocha when using the browser-runner in a subdirectory. 

It seems that Vite is not properly resolving the references to Mocha described in the template and this workaround fixes it. There may be a better solution for this problem that changes configuration in Vite to improve the module resolution, but I am not experienced enough with Vite to know where to start with that. This fix however, solves the immediate problem.

Fixes https://github.com/webdriverio/webdriverio/issues/11003

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browser-runner/src/vite/utils.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export async function getTemplate(options: WebdriverIO.BrowserRunnerOptions, env
         log.error(`Failed to setup source-map-support: ${(err as Error).message}`)
     }
 
+    const mochaPath = await resolve('mocha', `${rootFileUrl}/node_modules`);
+    const mochaCSSHref = path.join(url.fileURLToPath(path.dirname(mochaPath)), 'mocha.css');
+    const mochaJSSrc = path.join(url.fileURLToPath(path.dirname(mochaPath)), 'mocha.js');
```
Comment (christian-bromann) [inline packages/wdio-browser-runner/src/vite/utils.ts:59]:
```suggestion
    const mochaPath = await resolve('mocha', `${rootFileUrl}/node_modules`)
    const mochaCSSHref = path.join(url.fileURLToPath(path.dirname(mochaPath)), 'mocha.css')
    const mochaJSSrc = path.join(url.fileURLToPath(path.dirname(mochaPath)), 'mocha.js')
```

**Review by christian-bromann — COMMENTED** [review]
@FrederikBolding it seems this is failing in CI, Mocha can't be found, does it work locally for you?

**Comment by FrederikBolding** [comment]
> @FrederikBolding it seems this is failing in CI, Mocha can't be found, does it work locally for you?

Yes, this does work locally. I had forgotten to push up the updated snapshot, I just did that!

**Comment by FrederikBolding** [comment]
Looks like it only failed on Windows, perhaps some platform specific issue 🤔

**Comment by christian-bromann** [comment]
@FrederikBolding yes, this error:

> [0-0]  Error:  Test failed due to following error(s):
>  - mocha.js: Uncaught Error: Can't find Mocha attached to the `window` scope, was it installed? Run `npm install mocha` and run again!
    at http://localhost:50152/@fs/D:/a/webdriverio/webdriverio/packages/wdio-browser-runner/src/browser/frameworks/mocha.ts:12:11

basically says that it couldn't load the Mocha library. I recommend to use https://github.com/stateful/vscode-server-action and debug the case in CI (if you don't have a Windows machine).

**Comment by FrederikBolding** [comment]
@christian-bromann Latest commit seems to work locally on both Windows and Mac OS, I've also ran it through CI on my fork.

**Review by christian-bromann — APPROVED** [review]
Nice LGTM 👍

**Comment by axeleriksson147** [comment]
Hello! I had patched our `@wdio/browser-runner` 8.17.0 with this suggested fix without `@fs` and it was working:

```
<link rel="stylesheet" href="${mochaCSSHref}">
<script type="module" src="${mochaJSSrc}"></script>
```

with this version I get:

`Uncaught Error: Can't find Mocha attached to the `window` scope`

Just FYI. But we're in the inception of our component testing using these tools so maybe our config or setup is incorrect. @FrederikBolding @christian-bromann

**Comment by christian-bromann** [comment]
@axeleriksson147 can you share maybe a minimal reproducible example? Happy to take a look to get you up and running. BTW this patch is not yet released but from what I read from the comment you applied the changes manually?

**Comment by axeleriksson147** [comment]
@christian-bromann 🤦 🤦 I thought I saw this commit part of the `18.2.0` release but now I see it's part of `18.9.0` and it works perfectly. Thanks for the fix @FrederikBolding !

**Pull request merged**